crates/turborepo-lockfiles/src/pnpm/data.rs

```diff
@@ -375,6 +430,27 @@ impl crate::Lockfile for PnpmLockfile {
         name: &str,
         version: &str,
     ) -> Result<Option<crate::Package>, crate::Error> {
+        // Handle catalog references first
+        if version.starts_with("catalog:") {
+            let catalog_name = match version.strip_prefix("catalog:") {
+                Some("") => "default",
```

Comment:
Do we need to error on `"dep": "catalog:"`? I don't think inserting `default` as the version is correct.

Reply:
`catalog:` is how you reference the default catalog: https://pnpm.io/catalogs#the-catalog-protocol-catalog

`default` ends up being the item to look up in `self.catalogs`, mapping from the name that gets used in the lockfile. Can be seen in the fixture here: https://github.com/vercel/turborepo/blob/8a5880f100c842174a8f6e56049fe23c05dcdd24/crates/turborepo-lockfiles/fixtures/pnpm-catalog.yaml#L8

...At least that's how I'm understanding what I've written.